app/controllers/posts_controller.rb

```diff
@@ -460,6 +450,18 @@ def require_import_permission
     redirect_to new_post_path
   end
 
+  def require_edit_permission
+    return if @post.editable_by?(current_user) || @post.metadata_editable_by?(current_user)
+    flash[:error] = "You do not have permission to modify this post."
+    redirect_to @post
+  end
+
+  def require_delete_permission
+    return if @post.deletable_by?(current_user)
+    flash[:error] = "You do not have permission to modify this post."
+    redirect_to @post and return
```

Comment:
```suggestion
    redirect_to @post
```
(unnecessary - I know you removed this in another place :D)